Microsoft.Toolkit.Uwp.UI.Controls/RemoteDevicePicker/RemoteDevicePicker.cs

```diff
@@ -0,0 +1,358 @@
+
+        /// <summary>
+        /// Gets or sets the DeviceList Selection Mode. Defaults to ListViewSelectionMode.Single
+        /// </summary>
+        public RemoteDeviceSelectionMode SelectionMode
+        {
+            get { return (RemoteDeviceSelectionMode)GetValue(SelectionModeProperty); }
+            set { SetValue(SelectionModeProperty, value); }
+        }
+
+        /// <summary>
+        /// Gets the dependency property for <see cref="SelectionMode"/>.
+        /// </summary>
+        public static readonly DependencyProperty SelectionModeProperty = DependencyProperty.Register(
+            nameof(SelectionMode),
+            typeof(RemoteDeviceSelectionMode),
+            typeof(RemoteDevicePicker),
+            new PropertyMetadata(RemoteDeviceSelectionMode.Single));
+
+        /// <summary>
+        /// Gets or sets a value indicating whether Advanced Filters visible or not
+        /// </summary>
+        public bool ShowAdvancedFilters
+        {
+            get { return (bool)GetValue(ShowAdvancedFiltersProperty); }
+            set { SetValue(ShowAdvancedFiltersProperty, value); }
+        }
+
+        /// <summary>
+        /// Gets the dependency property for <see cref="ShowAdvancedFilters"/>.
+        /// </summary>
+        public static readonly DependencyProperty ShowAdvancedFiltersProperty = DependencyProperty.Register(
+            nameof(ShowAdvancedFilters),
+            typeof(bool),
+            typeof(RemoteDevicePicker),
+            new PropertyMetadata(false));
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RemoteDevicePicker"/> class.
+        /// </summary>
+        public RemoteDevicePicker()
+        {
+            DefaultStyleKey = typeof(RemoteDevicePicker);
+            RemoteSystems = new ObservableCollection<RemoteSystem>();
+            PrimaryButtonClick += RemoteDevicePicker_PrimaryButtonClick;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RemoteDevicePicker"/> class with filters.
+        /// </summary>
+        public RemoteDevicePicker(RemoteSystemDiscoveryType remoteSystemDiscoveryType, RemoteSystemAuthorizationKind remoteSystemAuthorizationKind, RemoteSystemStatusType remoteSystemStatusType)
+        {
+            DefaultStyleKey = typeof(RemoteDevicePicker);
+            RemoteSystems = new ObservableCollection<RemoteSystem>();
+            PrimaryButtonClick += RemoteDevicePicker_PrimaryButtonClick;
+
+            _discoveryFilter = new RemoteSystemDiscoveryTypeFilter(remoteSystemDiscoveryType);
+            _authorizationKindFilter = new RemoteSystemAuthorizationKindFilter(remoteSystemAuthorizationKind);
+            _statusFilter = new RemoteSystemStatusTypeFilter(remoteSystemStatusType);
+        }
+
+        private void RemoteDevicePicker_PrimaryButtonClick(ContentDialog sender, ContentDialogButtonClickEventArgs args)
+        {
+            _listDevices.SelectedItems.Clear();
```

Comment:
Primary is cancel (maybe clarify names with comments or something)?  Need null check here (or mode check).  I'm getting a 'catastrophic failure' crash when I'm in single selection mode and hit Cancel, this is the line.

Reply:
Fixed this.